### PR TITLE
Fix #20

### LINKED ISSUE #20: Network error running netease music
Hi, I got following error trying to play music in netease

image

Terminal log:

```
~ ❯ netease-cloud-music 
[1224/005307.771168:ERROR:cert_verify_proc_nss.cc(912)] CERT_PKIXVerifyCert for music.163.com failed err=-8181
[1224/005307.822616:ERROR:cert_verify_proc_nss.cc(912)] CERT_PKIXVerifyCert for sentry.music.163.com failed err=-8181
12-24, 00:53:10 [Error  ] [                          0] void netease::AudioPlayer::handleMediaStatusChanged(netease::VlcMediaPlayer::MediaStatus) Invalid Media: "1888622459_10_-bitrate-192-h3dSMF"
12-24, 00:53:10 [Error  ] [                          0] void netease::AudioPlayer::handleMediaStatusChanged(netease::VlcMediaPlayer::MediaStatus) Invalid Media: "1867217766_10_-bitrate-192-pNpBOZ"
```

Seems to be a cert error connecting to music.163.com but I can access it with curl or wget. I'm in KDE Plasma desktop environment.

Any ideas?

### PR DESCRIPTION
**Pull request by Freed-Wu**
Fix #20, network error

### CODE REVIEW

File: packages/netease-cloud-music/default.nix [hunk 1]
```diff
@@ -1,70 +1,445 @@
...
+    rm opt/netease/netease-cloud-music/libs/libUsageEnvironment.so.3
+    rm opt/netease/netease-cloud-music/libs/libxvidcore.so.4
+    rm opt/netease/netease-cloud-music/libs/libzvbi.so.0
+    rm opt/netease/netease-cloud-music/libs/libtdb.so.1
+    rm opt/netease/netease-cloud-music/libs/libtheoradec.so.1
+    rm opt/netease/netease-cloud-music/libs/libtheoraenc.so.1
+    rm opt/netease/netease-cloud-music/libs/libtwolame.so.0
+    rm opt/netease/netease-cloud-music/libs/libtheora.so.0
+    # rm opt/netease/netease-cloud-music/libs/libudev.so.1
+    # rm opt/netease/netease-cloud-music/libs/libXtst.so.6
```
Comment (linyinfeng) [inline packages/netease-cloud-music/default.nix]: Maybe these tons of `rm`s can be refactored to `rsync ... --include essential_so_pattern ...` in `installPhase`?

Comment (Freed-Wu) [reply packages/netease-cloud-music/default.nix]: I'll take a try.

**Comment by Freed-Wu** [comment]
Fixed

**Comment by Freed-Wu** [comment]
@linyinfeng

**Comment by linyinfeng** [comment]
We need a newer version of nixpkgs to support `curlOptsList`.

`nix run github:NixOS-CN/flakes/pull/51/head#netease-cloud-music`:
```
error:
       … while calling the 'listToAttrs' builtin

         at /nix/store/bfllm0im5z34n4v84zvip0kwy6vwl2yz-source/lib/attrsets.nix:139:5:

          138|   filterAttrsRecursive = pred: set:
          139|     listToAttrs (
             |     ^
          140|       concatMap (name:

       … while calling the 'concatMap' builtin

         at /nix/store/bfllm0im5z34n4v84zvip0kwy6vwl2yz-source/lib/attrsets.nix:140:7:

          139|     listToAttrs (
          140|       concatMap (name:
             |       ^
          141|         let v = set.${name}; in

       (stack trace truncated; use '--show-trace' to show the full trace)

       error: function 'anonymous lambda' called with unexpected argument 'curlOptsList'

       at /nix/store/bfllm0im5z34n4v84zvip0kwy6vwl2yz-source/pkgs/build-support/fetchurl/default.nix:40:1:

           39|
           40| { # URL to fetch.
             | ^
           41|   url ? ""
```

Seems the update workflow <https://github.com/NixOS-CN/flakes/blob/main/.github/workflows/update.yml> is broken since 2023-04-29 @Ninlives.

**Comment by Freed-Wu** [comment]
> We need a newer version of nixpkgs to support curlOptsList.

OK, I have changed `curlOptsList` to `curlOpts`.

**Comment by linyinfeng** [comment]
> > We need a newer version of nixpkgs to support curlOptsList.
> 
> OK, I have changed `curlOptsList` to `curlOpts`.

Actually, I mean, we need to fix the update workflow or do `nix flake update` manually instead of using the old nixpkgs...
And other packages will also need to be tested and fixed after an update.

But currently, [nixos-cn/flakes](https://github.com/NixOS-CN/flakes) is poorly maintained. @Ninlives even proposed that we should deprecate and archive it in [#nixos_zhcn:nichi.co](https://matrix.to/#/#zh-cn:nixos.org):
```
（2023-07-13T15:29+08:00）
mlatus (@Ninlives)
    我有点想把 nixos-cn/flakes archive 掉，实在没啥人维护（
    大家觉得呢
    感觉就去 nur 好了
Yinfeng
    维护度确实太低了
mlatus
    愿意维护就到 nixpkgs 去
    这个 flake 感觉比较鸡肋
    反正 sops-template 也 upstream 了
Yinfeng
    archive 掉得先把包整出去
mlatus
    有啥包大家还要的认领一下？
(然而之后并没有人认领，可见目前并没有人在维护这个仓库……）
```

So I recommend keeping the package in your own flake/nur. It needs some work to bring this repository back to normal, or in the worst situation, this repository will be deprecated and archived.

**Comment by Freed-Wu** [comment]
OK. Can this package be uploaded to nixos/nixpkgs? If not, I'll save this package in my NUR. When this repository have new changes, please ping me :smile:

**Comment by linyinfeng** [comment]
> OK. Can this package be uploaded to nixos/nixpkgs? If not, I'll save this package in my NUR. When this repository have new changes, please ping me smile

Yes, you can upstream it to nixpkgs.

**Pull request closed without merging**